scripts/capture_notimplemented_responses.py

```diff
@@ -57,16 +60,80 @@ def simulate_call(service: str, op: str) -> RowEntry:
         config=botocore.config.Config(
             parameter_validation=False,
             retries={"max_attempts": 0, "total_max_attempts": 1},
-            connect_timeout=50,
-            read_timeout=1,
+            connect_timeout=90,
+            read_timeout=120,
             inject_host_prefix=False,
         ),
     )
 
     service_model = service_models.get(service)
     op_model = service_model.operation_model(op)
-    parameters = generate_request(op_model)  # should be generate_parameters I guess
+    parameters = generate_request(op_model) or {}
+    result = _make_api_call(client, service, op, parameters)
+    error_msg = result.get("error_message", "")
+    if result.get("error_code", "") == "InternalError":
+        # some deeper investigation necessary, check for some common errors here and retry
+        if service == "apigateway" and "Unexpected HTTP method" in error_msg:
+            # moto currently raises exception in some requests, if the http method is not supported, meaning it is not implemented
+            result["error_code"] = 501  # reflect that this is not implemented
+        if (
+            "localstack.aws.protocol.parser.ProtocolParserError: Unable to parse request (not well-formed (invalid token)"
+            in error_msg
+        ):
+            # parsing errors might be due to invalid parameter values
+            # try to re-create params
+            logging.debug(
+                "ProtocolParserError detected, old parameters used: %s\nre-running request %s.%s with new parameters",
+                parameters,
+                service,
+                op,
+            )
+            parameters = generate_request(op_model) or {}
+            result = _make_api_call(client, service, op, parameters)
+        else:
+            while "TypeError" in error_msg and "got an unexpected keyword argument" in error_msg:
+                # sometimes LocalStack's autogenerated API is not yet up-to-date, which could cause
+                # if we see the 'unexpected keyword argument' error, we can re-try after removing the unknown argument
+                match = re.search("got an unexpected keyword argument '(.*)'", error_msg)
```

Comment:
oh, right - this still needs to be fixed. thanks for the hint! 
Currently it will break the loop because the keyword argument we are looking for, is not in the `parameters` dict anymore. It was intended to loop until the `TypeError` is gone. I will try to refactor this part, so it gets a bit clearer.